service/csv/pokemon.go

```diff
@@ -28,33 +77,95 @@ func GetPokemons() []model.Pokemon {
 	return pokemons
 }
 
+func (ps *PokemonService) SearchPokemonApi(name string) (string, error) {
+	res, err := http.Get("https://pokeapi.co/api/v2/pokemon/" + name)
+
+	if err != nil {
+		return "", err
+
+	}
+
+	res2, err2 := savePokemonCSV(res)
+
+	if err2 != nil {
+		return "", err2
+	}
+
+	return res2, nil
+
+}
+
+func savePokemonCSV(res *http.Response) (string, error) {
+
+	defer res.Body.Close()
+
+	content, error1 := ioutil.ReadAll(res.Body)
+
+	if error1 != nil {
+		return "", error1
+	}
+
+	var pokemon = new(model.Poke)
+
+	err2 := json.Unmarshal(content, &pokemon)
+
+	if err2 != nil {
+		return "", err2
+	}
+
+	file, err3 := os.OpenFile(parameters.CsvPath, os.O_APPEND|os.O_WRONLY, os.ModeAppend)
+	if err3 != nil {
+		return "", err3
+	}
+
+	defer file.Close()
+	var type1 string = "no type"
+	var type2 string = "no type"
+	if len(pokemon.Types) == 1 {
+		type1 = pokemon.Types[0].Type.Name
+	} else if len(pokemon.Types) == 2 {
+		type1 = pokemon.Types[0].Type.Name
+		type2 = pokemon.Types[1].Type.Name
+	}
```

Comment:
You could simplify logic here:

```suggestion
	type1 := pokemon.Types[0].Type.Name
	type2 := "no type"
	if len(pokemon.Types) == 2 {
		type2 = pokemon.Types[1].Type.Name
	}
```